mmtrack/models/sot/stark.py

```diff
@@ -0,0 +1,247 @@
+from collections import defaultdict
+
+import torch
+import torch.nn.functional as F
+from mmdet.models.builder import build_backbone, build_head, build_neck
+from torch.nn.modules.batchnorm import _BatchNorm
+from torch.nn.modules.conv import _ConvNd
+
+from ..builder import MODELS
+from .base import BaseSingleObjectTracker
+
+
+@MODELS.register_module()
+class Stark(BaseSingleObjectTracker):
+    """STARK: Learning Spatio-Temporal Transformer for Visual Tracking.
+
+    This single object tracker is the implementation of `STARk
+    <https://arxiv.org/abs/2103.17154>`_.
+
+    Args:
+        backbone (dict): the configuration of backbone network.
+        neck (dict, optional): the configuration of neck network.
+            Defaults to None.
+        head (dict, optional): the configuration of head network.
+            Defaults to None.
+        init_cfg (dict, optional): the configuration of initialization.
+            Defaults to None.
+        frozen_modules (str | list | tuple, optional): the names of frozen
+            modules. Defaults to None.
+        train_cfg (dict, optional): the configuratioin of train.
+            Defaults to None.
+        test_cfg (dict, optional): the configuration of test.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 backbone,
+                 neck=None,
+                 head=None,
+                 init_cfg=None,
+                 frozen_modules=None,
+                 train_cfg=None,
+                 test_cfg=None):
+        super(Stark, self).__init__(init_cfg)
+        self.backbone = build_backbone(backbone)
+        self.neck = build_neck(neck)
+        self.head = build_head(head)
+
+        self.test_cfg = test_cfg
+        self.train_cfg = train_cfg
+
+        # Set the update interval
+        self.update_intervals = self.test_cfg['update_intervals']
+        self.num_extra_template = len(self.update_intervals)
+
+        self.debug = False
+        if frozen_modules is not None:
+            self.freeze_module(frozen_modules)
+        # We only train classification head in stage-2. The transformer and
+        # bbox head are not trained in stage-2.
+        if self.head.run_cls_head and not self.head.run_bbox_head:
+            for name, module in self.head.named_parameters():
+                if 'cls_head' not in name:
+                    module.requires_grad = False
+
+    def init_weights(self):
+        """Initialize the weights of modules in single object tracker."""
+        # We don't use the `init_weights()` function in BaseModule, since it
+        # doesn't support the initialization method from `reset_parameters()`
+        # in Pytorch.
+        if self.with_backbone:
+            self.backbone.init_weights()
+
+        if self.with_neck:
+            for m in self.neck.modules():
+                if isinstance(m, _ConvNd) or isinstance(m, _BatchNorm):
+                    m.reset_parameters()
```

Comment:
Are you sure that the `neck` need to use `reset_parameters` to init ?

Reply:
I've checked it. It uses the default initialization method in the pytorch.